src/UXClient/Components/HierarchyNavigation/HierarchyNavigation.ts

```diff
@@ -856,12 +855,106 @@ class HierarchyNavigation extends Component{
+        contextMenuProps['eltMousePos'] = eltMousePos;
+        this.contextMenuProps = contextMenuProps;
+
+        this.clickedInstance = instanceObj;
+        instanceObj.node.classed('tsi-selected', true);
+    }
+
+    public drawContextMenu = (contextMenuItems, contextMenuOptions: ContextMenuOptions) => {
+        let itemList = [];
+        let contextMenuList;
+        this.contextMenu = this.contextMenuProps['target'].append('div').classed('tsi-hierarchyNavigationContextMenu', true).attr('style', () => `top: ${this.contextMenuProps['wrapperMousePos'] - this.contextMenuProps['eltMousePos']}px`);
+        let renderList = (contextMenuItems) => {
+            if (this.contextMenu.select("ul").empty()) {
+                contextMenuList = this.contextMenu.append('ul');
+            } else {
+                this.contextMenu.select("ul").text('');
+            }
+            
+            contextMenuItems.forEach(item => {
+                let option = item.name;
+                let li = contextMenuList.append('li');
+                
+                if (!contextMenuOptions.isSelectionEnabled) {
+                    li.attr('tabindex', 0)
+                    .attr('arialabel', option)
+                    .attr('title', option)
+                    .text(option).on('click keydown', () => {
+                        if (Utils.isKeyDownAndNotEnter(d3.event)) {return; }
+                        item.action();
+                    });
+                } else {
+                    li.on('click', function() {
+                        let elem = d3.select(this).select(".tsi-hierarchyCheckbox");
+                        if (elem.classed("tsi-notSelected")) {
+                            itemList.push(item);
+                            elem.classed("tsi-notSelected", false);
+                        } else {
+                            let index = itemList.map(elem => elem.name).indexOf(item.name);
+                            itemList.splice(index, 1);
+                            elem.classed("tsi-notSelected", true);
+                        }
+                    })
+                    let div = li.append('div').classed('tsi-selectionItemWrapper', true);
+                    div.append('span').classed('tsi-hierarchyCheckbox tsi-notSelected', true);
+                    div.append('span').classed('tsi-selectionItemKind', true).classed(item.kind, true);
+                    div.append('span').classed('tsi-selectionItem', true).text(option);
+                }
+            });
+        }
+
+        // draw filter box if enabled
+        if (contextMenuOptions.isFilterEnabled) {
+            let searchBox = this.contextMenu.append('div').classed('tsi-search', true);
+            searchBox.append('i').classed('tsi-search-icon', true);
+            searchBox.append('input').classed('tsi-searchInput', true).attr('placeholder', this.getString('Search'))
+                .on('input', function() { 
```

Comment:
so, the input box should be the full height and width of its parent div, with transparent background that allows the search icon to show through, then it has enough left padding that the text of the input box doesnt overlap the icon.  if you do that then you dont need to actual set that the parent is focused, because you can use the native focus styles of the input box

Reply:
this applies to all input boxes of this nature, e.g. autocomplete